crates/ra_assists/src/inline_local_variable.rs

```diff
@@ -0,0 +1,298 @@
+use hir::db::HirDatabase;
+use hir::source_binder::function_from_child_node;
+use ra_syntax::{ast::{self, AstNode}, TextRange};
+use ra_syntax::ast::{PatKind, ExprKind};
+
+use crate::{Assist, AssistCtx, AssistId};
+use crate::assist_ctx::AssistBuilder;
+
+pub(crate) fn inline_local_varialbe(mut ctx: AssistCtx<impl HirDatabase>) -> Option<Assist> {
+    let let_stmt = ctx.node_at_offset::<ast::LetStmt>()?;
+    let bind_pat = match let_stmt.pat()?.kind() {
+        PatKind::BindPat(pat) => pat,
+        _ => return None,
+    };
+    if bind_pat.is_mutable() {
+        return None;
+    }
+    let initializer = let_stmt.initializer()?;
+    let wrap_in_parens = match initializer.kind() {
+        ExprKind::LambdaExpr(_) => true,
+        ExprKind::IfExpr(_) => true,
+        ExprKind::LoopExpr(_) => true,
+        ExprKind::ForExpr(_) => true,
+        ExprKind::WhileExpr(_) => true,
+        ExprKind::ContinueExpr(_) => true,
+        ExprKind::BreakExpr(_) => true,
+        ExprKind::Label(_) => true,
+        ExprKind::ReturnExpr(_) => true,
+        ExprKind::MatchExpr(_) => true,
+        ExprKind::StructLit(_) => true,
+        ExprKind::CastExpr(_) => true,
+        ExprKind::PrefixExpr(_) => true,
+        ExprKind::RangeExpr(_) => true,
+        ExprKind::BinExpr(_) => true,
+        ExprKind::CallExpr(_) => false,
+        ExprKind::IndexExpr(_) => false,
+        ExprKind::MethodCallExpr(_) => false,
+        ExprKind::FieldExpr(_) => false,
+        ExprKind::TryExpr(_) => false,
+        ExprKind::RefExpr(_) => false,
+        ExprKind::Literal(_) => false,
+        ExprKind::TupleExpr(_) => false,
+        ExprKind::ArrayExpr(_) => false,
+        ExprKind::ParenExpr(_) => false,
+        ExprKind::PathExpr(_) => false,
+        ExprKind::BlockExpr(_) => false,
+    };
+
+    let delete_range = if let Some(whitespace) =
+        let_stmt.syntax().next_sibling().and_then(ast::Whitespace::cast)
+    {
+        TextRange::from_to(let_stmt.syntax().range().start(), whitespace.syntax().range().end())
+    } else {
+        let_stmt.syntax().range()
+    };
+
+    let init_str = if wrap_in_parens {
```

Comment:
Yeah, I imagine this could be smarter (we should look at the place where we are inserting the expression, and don't add `()` for if, etc), but this is a good start

Reply:
I'll send another PR when i am free.